optuna/storages/redis.py

```diff
@@ -0,0 +1,577 @@
+
+        self._check_study_id(study_id)
+
+        study_name_pkl = self.redis.get("study_id:{:010d}:study_name".format(study_id))
+        assert study_name_pkl is not None
+        return pickle.loads(study_name_pkl)
+
+    def get_study_direction(self, study_id):
+        # type: (int) -> structs.StudyDirection
+
+        direction_pkl = self.redis.get("study_id:{:010d}:direction".format(study_id))
+        assert direction_pkl is not None
+        return pickle.loads(direction_pkl)
+
+    def get_study_user_attrs(self, study_id):
+        # type: (int) -> Dict[str, Any]
+
+        with self._lock:
+            study_summary = self._get_study_summary(study_id)
+            return copy.deepcopy(study_summary.user_attrs)
+
+    def get_study_system_attrs(self, study_id):
+        # type: (int) -> Dict[str, Any]
+
+        with self._lock:
+            study_summary = self._get_study_summary(study_id)
+            return copy.deepcopy(study_summary.system_attrs)
+
+    @staticmethod
+    def _key_study_param_distribution(study_id):
+        # type: (int) -> str
+
+        return "study_id:{:010d}:params_distribution".format(study_id)
+
+    def _get_study_param_distribution(self, study_id):
+        # type: (int) -> Dict
+
+        if self.redis.exists(self._key_study_param_distribution(study_id)):
+            param_distribution_pkl = self.redis.get(self._key_study_param_distribution(study_id))
+            assert param_distribution_pkl is not None
+            return pickle.loads(param_distribution_pkl)
+        else:
+            return {}
+
+    def _set_study_param_distribution(self, study_id, param_distribution):
+        # type: (int, Dict) -> None
+
+        self.redis.set(self._key_study_param_distribution(study_id),
+                       pickle.dumps(param_distribution))
+
+    def get_all_study_summaries(self):
+        # type: () -> List[structs.StudySummary]
+
+        with self._lock:
+            study_summaries = []
+            study_ids = [pickle.loads(sid) for sid in self.redis.lrange("study_list", 0, -1)]
+            for study_id in study_ids:
+                study_summary = self._get_study_summary(study_id)
+                study_summaries.append(study_summary)
+
+        return study_summaries
+
+    def create_new_trial(self, study_id, template_trial=None):
+        # type: (int, Optional[structs.FrozenTrial]) -> int
+
+        self._check_study_id(study_id)
+
+        if template_trial is None:
+            trial = self._create_running_trial()
+        else:
+            trial = copy.deepcopy(template_trial)
+
+        if not self.redis.exists("trial_counter"):
+            self.redis.set("trial_counter", -1)
+
+        trial_id = self.redis.incr("trial_counter", 1)
+        trial_number = self.redis.incr("study_id:{:010d}:trial_number".format(study_id))
+        # TODO(pablete): Remove number from system_attrs after adding TrialModel.number.
+        trial.system_attrs['_number'] = trial_number
```

Comment:
Sorry could you remove this line? #939 was just merged 🙂

Reply:
Removed.